features/nanostack/nanostack-hal-mbed-cmsis-rtos/arm_hal_fhss_timer.cpp

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2018 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "ns_types.h"
+#include "fhss_api.h"
+#include "fhss_config.h"
+#include "mbed.h"
+#include "mbed_trace.h"
+#include "platform/arm_hal_interrupt.h"
+#include <Timer.h>
+
+#define TRACE_GROUP "fhdr"
+#ifndef NUMBER_OF_SIMULTANEOUS_TIMEOUTS
+#define NUMBER_OF_SIMULTANEOUS_TIMEOUTS  2
+#endif //NUMBER_OF_SIMULTANEOUS_TIMEOUTS
+
+static Timer timer;
```

Comment:
Follow-up job, as I failed to review earlier:

I would expect that `timer` and `fhss_timeout` would need to use `SingletonPtr` to avoid increasing static RAM size for people not using Nanostack. Otherwise they'll always be pulled in just to run the constructors.

Although SingletonPtr can't really handle an array :( Might end up having to just `new` `fhss_timeout`.

(Is there a CI job monitoring size increases for blinky or something?)

Reply:
I don't think so. I think the other @kegilbert, @studavekar, or @SenRamakri should know tho.